src/common/src/array/data_chunk.rs

```diff
@@ -150,7 +157,10 @@ impl DataChunk {
     }
 
     pub fn into_parts(self) -> (Vec<Column>, Option<Bitmap>) {
-        (self.columns, self.visibility)
+        match self.vis2 {
+            Vis::Bitmap(b) => (self.columns, Some(b)),
+            Vis::Compact(_) => (self.columns, None),
```

Comment:
`cardinality` info lost during this operation. This is why fixing #2663 requires refactoring of `into_parts` & `new` to preserve `cardinality` - but keeping both `visibility` & `cardinality` consistently and separately is hard. So we need this PR to pack them into a single `Vis`. Then `into_parts` & `new` can use `Vis` in the next PR.